src/Analyser/Scope.php

```diff
@@ -1120,7 +1094,15 @@ private function resolveType(Expr $node): Type
 			}
 			if ($constantClassType->hasConstant($constantName)) {
 				$constant = $constantClassType->getConstant($constantName);
-				return $this->getTypeFromValue($constant->getValue());
+				$constantType = $this->getTypeFromValue($constant->getValue());
+				$directClassNames = TypeUtils::getDirectClassNames($constantClassType);
+				if (
+					count($directClassNames) === 1 &&
+					in_array(sprintf('%s::%s', $directClassNames[0], $constantName), $this->dynamicConstantNames, true)
```

Comment:
You should ask the Broker for `$directClassNames[0]` class reflection and get the original class name to account for class aliases.